### ISSUE TITLE
Exit program due to configuration error?

### ISSUE THREAD

**Issue by clonejo** [jo-m/trainbot#13]
Considering this error case:
https://github.com/jo-m/trainbot/blob/38d8fc32d552a23863a63eecf56d25a948c9db9b/internal/pkg/stitch/auto.go#L237-L239

As far as i understand, this just comes down to the `--rect-w`, `--px-per-m` and `--max-speed-kph` parameters. So it's always a static configuration mistake. I propose exiting the program early with an error code.

**Comment by jo-m** [comment]
Yes, it would defn. make sense to exit early in this case. However, the later check is still needed, because we only know for 100% sure what resolution we get from the camera when we get it, so I'd like to keep it there as an assert.

**Comment by clonejo** [comment]
What about just panicking, like i did in bd40ae5? Then we also don't need the same check in a different place.

**Comment by jo-m** [comment]
I have placed the check where it currently is because this place in the code provides the context for the math in the check. If you move it somewhere else, this context is lost.

I am not 100% sure any more why I made the check to return an error instead of panicking. But in general, I tend to use panic only when something is a "bug", not user error. I think with some of the cameras I tested I sometimes received bad frames, and I did not want to crash, but just skip them and continue with the next one, and this error handling fits into that pattern.
